Cql/Cql.Compiler/ExpressionBuilder.cs

```diff
@@ -1,1186 +1,1277 @@
-
-            return new FunctionCallExpression(CqlExpressions.Definitions_PropertyExpression, libraryName, name, convertedArguments, definitionType);
-
-            bool CheckConversion(Type from, Type to) => _typeConverter.CanConvert(from, to);
-        }
-
-        protected Expression InvokeDefinitionThroughRuntimeContext(
-            string name,
-            string? libraryAlias,
-            LambdaExpression definition)
-        {
-            var type = definition.Type;
-            if (type.IsGenericType)
-            {
-                var typeArgs = type.GetGenericArguments();
-                var returnType = typeArgs[^1];
-                var invoke = InvokeDefinitionThroughRuntimeContext(name, libraryAlias, returnType);
-                return invoke;
-            }
-
-            throw this.NewExpressionBuildingException("LambdaExpressions should be a variant of Func<>");
-        }
-
-        protected Expression InvokeDefinitionThroughRuntimeContext(
-            string name,
-            string? libraryAlias,
-            Type definitionReturnType)
-        {
-            string libraryName = _libraryContext.GetNameAndVersionFromAlias(libraryAlias, throwError: false)
-                                 ?? throw this.NewExpressionBuildingException($"Local library {libraryAlias} is not defined; are you missing a using statement?");
-
-            var funcType = typeof(Func<,>).MakeGenericType(typeof(CqlContext), definitionReturnType);
-            return new DefinitionCallExpression(CqlExpressions.Definitions_PropertyExpression, libraryName, name, CqlExpressions.ParameterExpression, funcType);
-        }
-
-        private Expression ProcessValueSet(ValueSetRef valueSetRef, Elm.Expression valueExpr, bool isList)
-        {
-            Expression expr = Translate(valueExpr);
-
-            var codeType = isList ? _typeResolver.GetListElementType(expr.Type, throwError: true)! : expr.Type;
-
-            var valueSet = InvokeDefinitionThroughRuntimeContext(valueSetRef.name!, valueSetRef.libraryName, typeof(CqlValueSet));
-            if (codeType == _typeResolver.CodeType)
-                return BindCqlOperator(isList ? CqlOperator.CodesInValueSet : CqlOperator.CodeInValueSet, null, expr, valueSet);
-
-            if (codeType == _typeResolver.ConceptType)
-                return BindCqlOperator(isList ? CqlOperator.ConceptsInValueSet : CqlOperator.ConceptInValueSet, null, expr, valueSet);
-
-            if (codeType == typeof(string))
-                return BindCqlOperator(isList ? CqlOperator.StringsInValueSet: CqlOperator.StringInValueSet, null, expr, valueSet);
-
-            throw new NotImplementedException().WithContext(this);
-        }
-    }
+﻿/*
+ * Copyright (c) 2023, NCQA and contributors
+ * See the file CONTRIBUTORS for details.
+ *
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/firely-cql-sdk/main/LICENSE
+ */
+
+using System;
```

Comment:
This file has changed so much that I cannot review it. If I need to, we'll need to go over it together.

Reply:
Basically all the *-ExpressionBuilders were renamed to *-ExpressionBuilderContext. They were kept inside the same file for now and can be moved into a separate file during cleanup.  I added the new Builder types just above in the same flles. Probably, if I added those to the bottom of the file, GitHub would've had an easier job to do comparisons.